    Add C++ headers <barrier> <concepts> <execution> <latch> <numbers> <semaphore>
    
    I missed adding these to the libc++ Makefile, when importing
    llvm-project 11.0.0-rc1, even though they were supplied by upstream.
    
    While here, update OptionalObsoleteFiles.inc to add these new headers,
    and cleanup old cruft.
    
    Reported by:    yuri
    Submitted by:   jkim (Makefile diff)
    PR:             255374
    MFC after:      3 days
    
    (cherry picked from commit 95aa617e4bf09fcc813b1bab3d0dbf4b606807b1)
    (cherry picked from commit 70e13c4cffd5ff7a70296bc5c4c3b7525c278b1d)
--- lib/libc++/Makefile.orig
+++ lib/libc++/Makefile
@@ -112,6 +112,7 @@
 STD_HEADERS+=	any
 STD_HEADERS+=	array
 STD_HEADERS+=	atomic
+STD_HEADERS+=	barrier
 STD_HEADERS+=	bit
 STD_HEADERS+=	bitset
 STD_HEADERS+=	cassert
@@ -131,6 +132,7 @@
 STD_HEADERS+=	compare
 STD_HEADERS+=	complex
 STD_HEADERS+=	complex.h
+STD_HEADERS+=	concepts
 STD_HEADERS+=	condition_variable
 STD_HEADERS+=	csetjmp
 STD_HEADERS+=	csignal
@@ -149,6 +151,7 @@
 STD_HEADERS+=	deque
 STD_HEADERS+=	errno.h
 STD_HEADERS+=	exception
+STD_HEADERS+=	execution
 STD_HEADERS+=	fenv.h
 STD_HEADERS+=	filesystem
 STD_HEADERS+=	float.h
@@ -164,6 +167,7 @@
 STD_HEADERS+=	iostream
 STD_HEADERS+=	istream
 STD_HEADERS+=	iterator
+STD_HEADERS+=	latch
 STD_HEADERS+=	limits
 STD_HEADERS+=	limits.h
 STD_HEADERS+=	list
@@ -174,6 +178,7 @@
 STD_HEADERS+=	memory
 STD_HEADERS+=	mutex
 STD_HEADERS+=	new
+STD_HEADERS+=	numbers
 STD_HEADERS+=	numeric
 STD_HEADERS+=	optional
 STD_HEADERS+=	ostream
@@ -182,6 +187,7 @@
 STD_HEADERS+=	ratio
 STD_HEADERS+=	regex
 STD_HEADERS+=	scoped_allocator
+STD_HEADERS+=	semaphore
 STD_HEADERS+=	set
 STD_HEADERS+=	setjmp.h
 STD_HEADERS+=	shared_mutex
--- tools/build/mk/OptionalObsoleteFiles.inc.orig
+++ tools/build/mk/OptionalObsoleteFiles.inc
@@ -3831,6 +3831,7 @@
 OLD_FILES+=usr/include/c++/v1/any
 OLD_FILES+=usr/include/c++/v1/array
 OLD_FILES+=usr/include/c++/v1/atomic
+OLD_FILES+=usr/include/c++/v1/barrier
 OLD_FILES+=usr/include/c++/v1/bit
 OLD_FILES+=usr/include/c++/v1/bitset
 OLD_FILES+=usr/include/c++/v1/cassert
@@ -3850,6 +3851,7 @@
 OLD_FILES+=usr/include/c++/v1/compare
 OLD_FILES+=usr/include/c++/v1/complex
 OLD_FILES+=usr/include/c++/v1/complex.h
+OLD_FILES+=usr/include/c++/v1/concepts
 OLD_FILES+=usr/include/c++/v1/condition_variable
 OLD_FILES+=usr/include/c++/v1/csetjmp
 OLD_FILES+=usr/include/c++/v1/csignal
@@ -3869,14 +3871,12 @@
 OLD_FILES+=usr/include/c++/v1/deque
 OLD_FILES+=usr/include/c++/v1/errno.h
 OLD_FILES+=usr/include/c++/v1/exception
+OLD_FILES+=usr/include/c++/v1/execution
 OLD_FILES+=usr/include/c++/v1/experimental/__config
 OLD_FILES+=usr/include/c++/v1/experimental/__memory
 OLD_FILES+=usr/include/c++/v1/experimental/algorithm
-OLD_FILES+=usr/include/c++/v1/experimental/any
-OLD_FILES+=usr/include/c++/v1/experimental/chrono
 OLD_FILES+=usr/include/c++/v1/experimental/coroutine
 OLD_FILES+=usr/include/c++/v1/experimental/deque
-OLD_FILES+=usr/include/c++/v1/experimental/dynarray
 OLD_FILES+=usr/include/c++/v1/experimental/filesystem
 OLD_FILES+=usr/include/c++/v1/experimental/forward_list
 OLD_FILES+=usr/include/c++/v1/experimental/functional
@@ -3884,25 +3884,22 @@
 OLD_FILES+=usr/include/c++/v1/experimental/list
 OLD_FILES+=usr/include/c++/v1/experimental/map
 OLD_FILES+=usr/include/c++/v1/experimental/memory_resource
-OLD_FILES+=usr/include/c++/v1/experimental/numeric
-OLD_FILES+=usr/include/c++/v1/experimental/optional
 OLD_FILES+=usr/include/c++/v1/experimental/propagate_const
-OLD_FILES+=usr/include/c++/v1/experimental/ratio
 OLD_FILES+=usr/include/c++/v1/experimental/regex
 OLD_FILES+=usr/include/c++/v1/experimental/set
 OLD_FILES+=usr/include/c++/v1/experimental/simd
 OLD_FILES+=usr/include/c++/v1/experimental/string
-OLD_FILES+=usr/include/c++/v1/experimental/string_view
-OLD_FILES+=usr/include/c++/v1/experimental/system_error
-OLD_FILES+=usr/include/c++/v1/experimental/tuple
 OLD_FILES+=usr/include/c++/v1/experimental/type_traits
 OLD_FILES+=usr/include/c++/v1/experimental/unordered_map
 OLD_FILES+=usr/include/c++/v1/experimental/unordered_set
 OLD_FILES+=usr/include/c++/v1/experimental/utility
 OLD_FILES+=usr/include/c++/v1/experimental/vector
+OLD_DIRS+=usr/include/c++/v1/experimental
 OLD_FILES+=usr/include/c++/v1/ext/__hash
 OLD_FILES+=usr/include/c++/v1/ext/hash_map
 OLD_FILES+=usr/include/c++/v1/ext/hash_set
+OLD_DIRS+=usr/include/c++/v1/ext
+OLD_FILES+=usr/include/c++/v1/fenv.h
 OLD_FILES+=usr/include/c++/v1/filesystem
 OLD_FILES+=usr/include/c++/v1/float.h
 OLD_FILES+=usr/include/c++/v1/forward_list
@@ -3917,6 +3914,7 @@
 OLD_FILES+=usr/include/c++/v1/iostream
 OLD_FILES+=usr/include/c++/v1/istream
 OLD_FILES+=usr/include/c++/v1/iterator
+OLD_FILES+=usr/include/c++/v1/latch
 OLD_FILES+=usr/include/c++/v1/limits
 OLD_FILES+=usr/include/c++/v1/limits.h
 OLD_FILES+=usr/include/c++/v1/list
@@ -3927,7 +3925,7 @@
 OLD_FILES+=usr/include/c++/v1/memory
 OLD_FILES+=usr/include/c++/v1/mutex
 OLD_FILES+=usr/include/c++/v1/new
-OLD_FILES+=usr/include/c++/v1/numeric
+OLD_FILES+=usr/include/c++/v1/numbers
 OLD_FILES+=usr/include/c++/v1/numeric
 OLD_FILES+=usr/include/c++/v1/optional
 OLD_FILES+=usr/include/c++/v1/ostream
@@ -3936,6 +3934,7 @@
 OLD_FILES+=usr/include/c++/v1/ratio
 OLD_FILES+=usr/include/c++/v1/regex
 OLD_FILES+=usr/include/c++/v1/scoped_allocator
+OLD_FILES+=usr/include/c++/v1/semaphore
 OLD_FILES+=usr/include/c++/v1/set
 OLD_FILES+=usr/include/c++/v1/setjmp.h
 OLD_FILES+=usr/include/c++/v1/shared_mutex
@@ -3956,12 +3955,12 @@
 OLD_FILES+=usr/include/c++/v1/system_error
 OLD_FILES+=usr/include/c++/v1/tgmath.h
 OLD_FILES+=usr/include/c++/v1/thread
-OLD_FILES+=usr/include/c++/v1/version
 OLD_FILES+=usr/include/c++/v1/tr1/__bit_reference
 OLD_FILES+=usr/include/c++/v1/tr1/__bsd_locale_defaults.h
 OLD_FILES+=usr/include/c++/v1/tr1/__bsd_locale_fallbacks.h
 OLD_FILES+=usr/include/c++/v1/tr1/__config
 OLD_FILES+=usr/include/c++/v1/tr1/__debug
+OLD_FILES+=usr/include/c++/v1/tr1/__errc
 OLD_FILES+=usr/include/c++/v1/tr1/__functional_03
 OLD_FILES+=usr/include/c++/v1/tr1/__functional_base
 OLD_FILES+=usr/include/c++/v1/tr1/__functional_base_03
@@ -3969,6 +3968,7 @@
 OLD_FILES+=usr/include/c++/v1/tr1/__libcpp_version
 OLD_FILES+=usr/include/c++/v1/tr1/__locale
 OLD_FILES+=usr/include/c++/v1/tr1/__mutex_base
+OLD_FILES+=usr/include/c++/v1/tr1/__node_handle
 OLD_FILES+=usr/include/c++/v1/tr1/__nullptr
 OLD_FILES+=usr/include/c++/v1/tr1/__split_buffer
 OLD_FILES+=usr/include/c++/v1/tr1/__sso_allocator
@@ -3982,6 +3982,8 @@
 OLD_FILES+=usr/include/c++/v1/tr1/any
 OLD_FILES+=usr/include/c++/v1/tr1/array
 OLD_FILES+=usr/include/c++/v1/tr1/atomic
+OLD_FILES+=usr/include/c++/v1/tr1/barrier
+OLD_FILES+=usr/include/c++/v1/tr1/bit
 OLD_FILES+=usr/include/c++/v1/tr1/bitset
 OLD_FILES+=usr/include/c++/v1/tr1/cassert
 OLD_FILES+=usr/include/c++/v1/tr1/ccomplex
@@ -3989,6 +3991,7 @@
 OLD_FILES+=usr/include/c++/v1/tr1/cerrno
 OLD_FILES+=usr/include/c++/v1/tr1/cfenv
 OLD_FILES+=usr/include/c++/v1/tr1/cfloat
+OLD_FILES+=usr/include/c++/v1/tr1/charconv
 OLD_FILES+=usr/include/c++/v1/tr1/chrono
 OLD_FILES+=usr/include/c++/v1/tr1/cinttypes
 OLD_FILES+=usr/include/c++/v1/tr1/ciso646
@@ -3996,8 +3999,10 @@
 OLD_FILES+=usr/include/c++/v1/tr1/clocale
 OLD_FILES+=usr/include/c++/v1/tr1/cmath
 OLD_FILES+=usr/include/c++/v1/tr1/codecvt
+OLD_FILES+=usr/include/c++/v1/tr1/compare
 OLD_FILES+=usr/include/c++/v1/tr1/complex
 OLD_FILES+=usr/include/c++/v1/tr1/complex.h
+OLD_FILES+=usr/include/c++/v1/tr1/concepts
 OLD_FILES+=usr/include/c++/v1/tr1/condition_variable
 OLD_FILES+=usr/include/c++/v1/tr1/csetjmp
 OLD_FILES+=usr/include/c++/v1/tr1/csignal
@@ -4016,6 +4021,9 @@
 OLD_FILES+=usr/include/c++/v1/tr1/deque
 OLD_FILES+=usr/include/c++/v1/tr1/errno.h
 OLD_FILES+=usr/include/c++/v1/tr1/exception
+OLD_FILES+=usr/include/c++/v1/tr1/execution
+OLD_FILES+=usr/include/c++/v1/tr1/fenv.h
+OLD_FILES+=usr/include/c++/v1/tr1/filesystem
 OLD_FILES+=usr/include/c++/v1/tr1/float.h
 OLD_FILES+=usr/include/c++/v1/tr1/forward_list
 OLD_FILES+=usr/include/c++/v1/tr1/fstream
@@ -4029,6 +4037,7 @@
 OLD_FILES+=usr/include/c++/v1/tr1/iostream
 OLD_FILES+=usr/include/c++/v1/tr1/istream
 OLD_FILES+=usr/include/c++/v1/tr1/iterator
+OLD_FILES+=usr/include/c++/v1/tr1/latch
 OLD_FILES+=usr/include/c++/v1/tr1/limits
 OLD_FILES+=usr/include/c++/v1/tr1/limits.h
 OLD_FILES+=usr/include/c++/v1/tr1/list
@@ -4039,7 +4048,7 @@
 OLD_FILES+=usr/include/c++/v1/tr1/memory
 OLD_FILES+=usr/include/c++/v1/tr1/mutex
 OLD_FILES+=usr/include/c++/v1/tr1/new
-OLD_FILES+=usr/include/c++/v1/tr1/numeric
+OLD_FILES+=usr/include/c++/v1/tr1/numbers
 OLD_FILES+=usr/include/c++/v1/tr1/numeric
 OLD_FILES+=usr/include/c++/v1/tr1/optional
 OLD_FILES+=usr/include/c++/v1/tr1/ostream
@@ -4048,9 +4057,11 @@
 OLD_FILES+=usr/include/c++/v1/tr1/ratio
 OLD_FILES+=usr/include/c++/v1/tr1/regex
 OLD_FILES+=usr/include/c++/v1/tr1/scoped_allocator
+OLD_FILES+=usr/include/c++/v1/tr1/semaphore
 OLD_FILES+=usr/include/c++/v1/tr1/set
 OLD_FILES+=usr/include/c++/v1/tr1/setjmp.h
 OLD_FILES+=usr/include/c++/v1/tr1/shared_mutex
+OLD_FILES+=usr/include/c++/v1/tr1/span
 OLD_FILES+=usr/include/c++/v1/tr1/sstream
 OLD_FILES+=usr/include/c++/v1/tr1/stack
 OLD_FILES+=usr/include/c++/v1/tr1/stdbool.h
@@ -4077,8 +4088,10 @@
 OLD_FILES+=usr/include/c++/v1/tr1/valarray
 OLD_FILES+=usr/include/c++/v1/tr1/variant
 OLD_FILES+=usr/include/c++/v1/tr1/vector
+OLD_FILES+=usr/include/c++/v1/tr1/version
 OLD_FILES+=usr/include/c++/v1/tr1/wchar.h
 OLD_FILES+=usr/include/c++/v1/tr1/wctype.h
+OLD_DIRS+=usr/include/c++/v1/tr1
 OLD_FILES+=usr/include/c++/v1/tuple
 OLD_FILES+=usr/include/c++/v1/type_traits
 OLD_FILES+=usr/include/c++/v1/typeindex
@@ -4092,8 +4105,10 @@
 OLD_FILES+=usr/include/c++/v1/valarray
 OLD_FILES+=usr/include/c++/v1/variant
 OLD_FILES+=usr/include/c++/v1/vector
+OLD_FILES+=usr/include/c++/v1/version
 OLD_FILES+=usr/include/c++/v1/wchar.h
 OLD_FILES+=usr/include/c++/v1/wctype.h
+OLD_DIRS+=usr/include/c++/v1
 OLD_FILES+=usr/lib32/libc++.a
 OLD_FILES+=usr/lib32/libc++.so
 OLD_LIBS+=usr/lib32/libc++.so.1
@@ -4104,10 +4119,6 @@
 OLD_FILES+=usr/lib32/libcxxrt.so
 OLD_LIBS+=usr/lib32/libcxxrt.so.1
 OLD_FILES+=usr/lib32/libcxxrt_p.a
-OLD_DIRS+=usr/include/c++/v1/tr1
-OLD_DIRS+=usr/include/c++/v1/experimental
-OLD_DIRS+=usr/include/c++/v1/ext
-OLD_DIRS+=usr/include/c++/v1
 .endif
 
 .if ${MK_LLD} == no
